src/cargo/ops/cargo_new.rs

```diff
@@ -925,3 +941,79 @@ fn update_manifest_with_inherited_workspace_package_keys(
         try_remove_and_inherit_package_key(key, manifest);
     }
 }
+
+/// Adds the new package member to the [workspace.members] array.
+/// - It first checks if the name matches any element in [workspace.exclude],
+///  and it ignores the name if there is a match.
+/// - Then it check if the name matches any element already in [workspace.members],
+/// and it ignores the name if there is a match.
+/// - If [workspace.members] doesn't exist in the manifest, the new package
+/// is not added to the members array. This prevents manifest files from breaking
+/// if they're only used for package templating.
```

Comment:
This isn't making sense to me for why we would do this; this is still allowing us to create packages that will fail otherwise.

There might be tests that operate on these failed packages but that was a stop gap until this issue was resolved.